apps/dot_ext/forms.py

```diff
@@ -100,17 +106,6 @@ def clean_name(self):
                                         """)
         return name
 
-    def clean_client_type(self):
-        client_type = self.cleaned_data.get('client_type')
-        authorization_grant_type = self.cleaned_data.get(
-            'authorization_grant_type')
-        if client_type == 'public' and authorization_grant_type == 'authorization-code':
-            msg = _(
-                'A public client may not request an '
-                'authorization-code grant type.')
-            raise forms.ValidationError(msg)
-        return client_type
-
```

Comment:
Removing this method, since this validation is already being performed in the `clean()` method.